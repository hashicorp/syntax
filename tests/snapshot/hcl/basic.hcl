# line comment

// line comment

/*
  Block comment
*/

io_mode = "async"

service "http" "web_proxy" {
  listen_addr = "127.0.0.1:8080"

  process "main" {
    command = ["/usr/local/bin/awesome-app", "server"]
  }

  process "mgmt" {
    command = ["/usr/local/bin/awesome-app", "mgmt"]
  }
}

sum = 1 + addend

message = "Hello, ${name}!"

shouty_message = upper(message)
