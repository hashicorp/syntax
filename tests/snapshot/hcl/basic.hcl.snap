># line comment
#^ source.hcl comment.line.number-sign.hcl punctuation.definition.comment.hcl
# ^^^^^^^^^^^^^ source.hcl comment.line.number-sign.hcl
>
>// line comment
#^^ source.hcl comment.line.double-slash.hcl punctuation.definition.comment.hcl
#  ^^^^^^^^^^^^^ source.hcl comment.line.double-slash.hcl
>
>/*
#^^ source.hcl comment.block.hcl punctuation.definition.comment.hcl
>  Block comment
#^^^^^^^^^^^^^^^^ source.hcl comment.block.hcl
>*/
#^^ source.hcl comment.block.hcl punctuation.definition.comment.hcl
>
>io_mode = "async"
#^^^^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#       ^ source.hcl variable.declaration.hcl
#        ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#         ^ source.hcl variable.declaration.hcl
#          ^ source.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#           ^^^^^ source.hcl string.quoted.double.hcl
#                ^ source.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>
>service "http" "web_proxy" {
#^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#       ^ source.hcl meta.block.hcl
#        ^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#              ^ source.hcl meta.block.hcl
#               ^^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                          ^ source.hcl meta.block.hcl
#                           ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>  listen_addr = "127.0.0.1:8080"
#^^ source.hcl meta.block.hcl
#  ^^^^^^^^^^^ source.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#             ^ source.hcl meta.block.hcl variable.declaration.hcl
#              ^ source.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#               ^ source.hcl meta.block.hcl variable.declaration.hcl
#                ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                 ^^^^^^^^^^^^^^ source.hcl meta.block.hcl string.quoted.double.hcl
#                               ^ source.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>
>  process "main" {
#^^ source.hcl meta.block.hcl
#  ^^^^^^^ source.hcl meta.block.hcl meta.block.hcl entity.name.type.hcl
#         ^ source.hcl meta.block.hcl meta.block.hcl
#          ^^^^^^ source.hcl meta.block.hcl meta.block.hcl variable.other.enummember.hcl
#                ^ source.hcl meta.block.hcl meta.block.hcl
#                 ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.block.begin.hcl
>    command = ["/usr/local/bin/awesome-app", "server"]
#^^^^ source.hcl meta.block.hcl meta.block.hcl
#    ^^^^^^^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#           ^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl
#            ^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#             ^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl
#              ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.brackets.begin.hcl
#               ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl
#                                          ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
#                                           ^ source.hcl meta.block.hcl meta.block.hcl punctuation.separator.hcl
#                                            ^ source.hcl meta.block.hcl meta.block.hcl
#                                             ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                                              ^^^^^^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl
#                                                    ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
#                                                     ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.brackets.end.hcl
>  }
#^^ source.hcl meta.block.hcl meta.block.hcl
#  ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>  process "mgmt" {
#^^ source.hcl meta.block.hcl
#  ^^^^^^^ source.hcl meta.block.hcl meta.block.hcl entity.name.type.hcl
#         ^ source.hcl meta.block.hcl meta.block.hcl
#          ^^^^^^ source.hcl meta.block.hcl meta.block.hcl variable.other.enummember.hcl
#                ^ source.hcl meta.block.hcl meta.block.hcl
#                 ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.block.begin.hcl
>    command = ["/usr/local/bin/awesome-app", "mgmt"]
#^^^^ source.hcl meta.block.hcl meta.block.hcl
#    ^^^^^^^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl variable.other.readwrite.hcl
#           ^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl
#            ^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#             ^ source.hcl meta.block.hcl meta.block.hcl variable.declaration.hcl
#              ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.brackets.begin.hcl
#               ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl
#                                          ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
#                                           ^ source.hcl meta.block.hcl meta.block.hcl punctuation.separator.hcl
#                                            ^ source.hcl meta.block.hcl meta.block.hcl
#                                             ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#                                              ^^^^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl
#                                                  ^ source.hcl meta.block.hcl meta.block.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
#                                                   ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.brackets.end.hcl
>  }
#^^ source.hcl meta.block.hcl meta.block.hcl
#  ^ source.hcl meta.block.hcl meta.block.hcl punctuation.section.block.end.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>sum = 1 + addend
#^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#   ^ source.hcl variable.declaration.hcl
#    ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#     ^ source.hcl variable.declaration.hcl
#      ^ source.hcl constant.numeric.integer.hcl
#       ^ source.hcl
#        ^ source.hcl keyword.operator.arithmetic.hcl
#         ^^^^^^^^ source.hcl
>
>message = "Hello, ${name}!"
#^^^^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#       ^ source.hcl variable.declaration.hcl
#        ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#         ^ source.hcl variable.declaration.hcl
#          ^ source.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
#           ^^^^^^ source.hcl string.quoted.double.hcl
#                 ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl
#                  ^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.other.interpolation.begin.hcl
#                    ^^^^ source.hcl string.quoted.double.hcl meta.interpolation.hcl variable.other.readwrite.hcl
#                        ^ source.hcl string.quoted.double.hcl meta.interpolation.hcl keyword.other.interpolation.end.hcl
#                         ^ source.hcl string.quoted.double.hcl
#                          ^ source.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
>
>shouty_message = upper(message)
#^^^^^^^^^^^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#              ^ source.hcl variable.declaration.hcl
#               ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#                ^ source.hcl variable.declaration.hcl
#                 ^^^^^ source.hcl meta.function-call.hcl variable.function.hcl
#                      ^ source.hcl meta.function-call.hcl punctuation.section.parens.begin.hcl
#                       ^^^^^^^ source.hcl meta.function-call.hcl
#                              ^ source.hcl meta.function-call.hcl punctuation.section.parens.end.hcl
>